### PR TITLE
[develop] Add radar reflectivity task.

### LINKED ISSUE #489: Radar Reflectivity observations processing task is needed for RRFS
## Description
RRFS runs a pre-processing task on MRMS radar reflectivity files. Specifically the process_radarref task executes `process_NSSL_mosaic.exe` from rrfs-utl on files like `MergedReflectivityQC_00.50_${YYYY}${MM}${DD}-${HH}${min}${ss}.grib2`.

## Solution
Use the RRFS_dev1 version of this task as a guide to a solution. Include any necessary default configuration settings, add j-jobs, and ex-scripts, and hook them into the workflow.

This task has no dependencies other than data retrieval, so should be able to stand alone as a full workflow without other components.

## Requirements**

- All contributions must be machine independent and follow the style and guidelines of the SRW App.
- The contributed scripts must be tested, and new tests should be contributed for SRW. These WE2E tests can be short and only include data retrieval and running the task.

## Acceptance Criteria (Definition of Done)
Matches answers obtained in RRFS. Please reach out to someone at NOAA GSL to help with this test case.

### PR DESCRIPTION
**Pull request by danielabdi-noaa**
## DESCRIPTION OF CHANGES: 

Add `process_radarref` task from `RRFS_dev1`.
Note that there is no workflow so this PR can not be run independently until then.
However, it did successfully run for both spinup/production cycles in PR #540.

### Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## TESTS CONDUCTED: 

- [X] hera.intel
  The CONUS 3km test case run successfully with this task in PR #540 
- [ ] orion.intel
- [ ] cheyenne.intel
- [ ] cheyenne.gnu
- [ ] gaea.intel
- [ ] jet.intel
- [ ] wcoss2.intel
- [ ] NOAA Cloud (indicate which platform)
- [ ] Jenkins
- [ ] fundamental test suite
- [ ] comprehensive tests (specify *which* if a subset was used)

## ISSUE: 

- #489 

## CHECKLIST

- [X] My code follows the style guidelines in the Contributor's Guide
- [ ] I have performed a self-review of my own code using the Code Reviewer's Guide
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] My changes need updates to the documentation. I have made corresponding changes to the documentation
- [ ] My changes do not require updates to the documentation (explain).
- [ ] My changes generate no new warnings
- [ ] New and existing tests pass with my changes
- [ ] Any dependent changes have been merged and published

## LABELS (optional): 

A Code Manager needs to add the following labels to this PR: 
- [ ] Work In Progress
- [ ] bug
- [ ] enhancement
- [ ] documentation
- [ ] release
- [ ] high priority
- [ ] run_ci
- [ ] run_we2e_fundamental_tests
- [ ] run_we2e_comprehensive_tests
- [ ] Needs Cheyenne test 
- [ ] Needs Jet test 
- [ ] Needs Hera test 
- [ ] Needs Orion test 
- [ ] help wanted

### CODE REVIEW

**Review by panll — APPROVED** [review]
Tested on Hera. It works fine.

File: jobs/JREGIONAL_PROCESS_BUFR
Comment (EdwardSnyder-NOAA) [inline jobs/JREGIONAL_PROCESS_BUFR]: Looks like the BUFR jobs script was added instead of the Radar Ref.

Comment (danielabdi-noaa) [reply jobs/JREGIONAL_PROCESS_BUFR]: Thank a lot, I have replaced the j-job with the right one!

File: scripts/exregional_process_radarref.sh
Comment (christinaholtNOAA) [inline scripts/exregional_process_radarref.sh]: Please remove commented code here and for gridspec_dir above.

File: jobs/JREGIONAL_PROCESS_RADARREF
Comment (christinaholtNOAA) [inline jobs/JREGIONAL_PROCESS_RADARREF:9]: Please add the required environment variables coming from Rocoto in the description here.

File: scripts/exregional_process_radarref.sh
Comment (christinaholtNOAA) [inline scripts/exregional_process_radarref.sh]: It would be more efficient here and below to just check to see if ${HH} is contained in the array. For example:

```

if [[ ${CYCL_HRS_SPINSTART[@]} =~ "$HH" ]] ; then
  BKTYPE=1
fi
```

File: scripts/exregional_process_radarref.sh
Comment (christinaholtNOAA) [inline scripts/exregional_process_radarref.sh:134]: Removing the random large white space in the cp_vrfy statements is preferable. Also, this is the only use of list_iolayout, so making the computation closer to where it's used is easier to understand. It's potentially cleaner to go with:

```
for iii in $(seq -w 0 $(printf %4.4i $n_iolayouty))
do 
  cp_vrfy ....
done
```

File: scripts/exregional_process_radarref.sh
Comment (christinaholtNOAA) [inline scripts/exregional_process_radarref.sh]: We do want to make sure that all the machine case statements are removed from the scripts.

File: scripts/exregional_process_radarref.sh
Comment (christinaholtNOAA) [inline scripts/exregional_process_radarref.sh:178]: Perhaps this could just be an else?

File: scripts/exregional_process_radarref.sh
Comment (christinaholtNOAA) [inline scripts/exregional_process_radarref.sh]: bigmin is very difficult to understand from a readability perspective. Maybe the variable name `timelevel` would be clearer given the description of that variable? At least them we can recall that we're looping through the RADARREFL_TIMELEVEL list.

File: scripts/exregional_process_radarref.sh
Comment (christinaholtNOAA) [inline scripts/exregional_process_radarref.sh:155]: Maybe add a comment here that this internal loop is looking for files within close proximity to the time level in question?

File: scripts/exregional_process_radarref.sh
Comment (christinaholtNOAA) [inline scripts/exregional_process_radarref.sh]: Perhaps `sec` or `second` here to be consistent with `min`? It's hard to visually pick out individial s's in this chunk of code.

File: ush/machine/jet.yaml
Comment (christinaholtNOAA) [inline ush/machine/jet.yaml]: Do we want to make this change? We need to keep the paths pointing to where the EPIC-supported data for SRW lives and ensure that's populated with the right data.

File: ush/machine/jet.yaml
Comment (christinaholtNOAA) [inline ush/machine/jet.yaml]: We definitely do not want this path to be a SRW community dependency.

**Review by christinaholtNOAA — CHANGES_REQUESTED** [review]
I left several suggestions for clean up below, but we definitely need to make sure that we aren't pointing to unsupported data paths before merging.

File: ush/machine/jet.yaml
Comment (EdwardSnyder-NOAA) [reply ush/machine/jet.yaml]: @christinaholtNOAA - [PR 674](https://github.com/ufs-community/ufs-srweather-app/pull/674) switches the UFS_SRW_App paths to EPIC locations from the SRW role account locations. We've done testing to ensure experiments ran successfully.

File: ush/config_defaults.yaml
Comment (christinaholtNOAA) [inline ush/config_defaults.yaml]: This should probably be `OBS_SUFFIX`.

**Review by christinaholtNOAA — COMMENTED** [review]
One more tiny change request, but the changes look good. 

It looks like the supported radar obs paths are here:

Jet: /public/data/radar/mrms
Hera: /scratch2/BMC/public/data/radar/nssl/mrms/conus

@mkavulich Might find this useful for his work with data retrieval, too.

**Comment by danielabdi-noaa** [comment]
> One more tiny change request, but the changes look good.
> 
> It looks like the supported radar obs paths are here:
> 
> Jet: /public/data/radar/mrms Hera: /scratch2/BMC/public/data/radar/nssl/mrms/conus
> 
> @mkavulich Might find this useful for his work with data retrieval, too.

@christinaholtNOAA  Note that those are paths for realtime forecasts only. You can find the full list of observation paths for realtime/retro runs in this commit for hera/jet/wcoss2: [obs data paths](https://github.com/ufs-community/ufs-srweather-app/pull/647/commits/db8524fb80d421000f8723eb791c90396909d3c8)

**Review by christinaholtNOAA — APPROVED** [review]
LGTM.

**Comment by christinaholtNOAA** [comment]
@MichaelLueken In our RRFS dev merge discussion yesterday we chatted about pushing this one through when it's ready. If we can go ahead with that, it would be helpful. Thanks!

**Comment by MichaelLueken** [comment]
@danielabdi-noaa I'm working on reviewing and testing this PR now.  It looks like there is a conflict in `ush/config_defaults.yaml`.  Please merge develop into your `feature/task_radarref` branch one more time and I will approve this work and kick off the Jenkins tests.  Thanks!

**Comment by danielabdi-noaa** [comment]
@MichaelLueken I have resolved the conflicts. You can kick off the Jenkins tests now.

**Review by MichaelLueken — APPROVED** [review]
@danielabdi-noaa These changes look good to me!  The necessary components to run this task and the process bufr task will be completed in PR #725.  Approving now.

**Comment by MichaelLueken** [comment]
The Jenkins tests have completed.  The Hera Intel tests failed.  Manually running them, all tests successfully passed (with the exception of `grid_RRFS_CONUS_25km_ics_FV3GFS_lbcs_RAP_suite_HRRR`, which failed due to issue #688).  Moving forward with merging this work now.

**Pull request merged**
